compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -786,6 +794,7 @@ impl<'a> Parser<'a> {
             || self.check(&token::Not)
             || self.check(&token::Question)
             || self.check(&token::Tilde)
+            || self.check_keyword(kw::Const)
```

Comment:
I need to investigate if this visibly changes the MBE matching behavior for stable users. I hope not since this check is required for correctness.

Reply:
Doesn't seem to be a problem. I've added a test to ensure that `const Trait` doesn't regress stable code: tests/ui/rfcs/rfc-2632-const-trait-impl/mbe-bare-trait-objects-const-trait-bounds.rs

Reply:
It's obvious in hindsight but this doesn't regress the aforementioned test because I haven't (and I won't) add `kw::Const` & `TokenKind::Tilde` to `can_begin_type`.